src/lazy/encoder/write_as_ion.rs

```diff
@@ -151,7 +151,7 @@ macro_rules! impl_write_as_ion_value_for_iterable {
             for<'a> &'a $item: WriteAsIon,
         {
             fn write_as_ion_value<V: ValueWriter>(&self, writer: V) -> IonResult<()> {
-                writer.write_list(|list| {
+                writer.write_list(|list: &mut V::ListWriter| {
```

Comment:
Is this declaration of type required? I didn't see it in other places e.g. `write_sexp`.

Reply:
Good eye--there was a brief time during development when it was needed but it's no longer necessary. I'll remove it.